src/useForm.tsx

```diff
@@ -135,15 +138,16 @@ function createFields<T>(
             )
           },
           validate: () => {
-            setState(currentState =>
-              produce(currentState, updatedState => {
+            setState(currentState => {
+              const fieldValues = extractValuesFromFieldsState(currentState)
```

Comment:
Same change as above, we're interested in the form values themselves rather than extra metadata about them.